src/main/scala/sigmastate/utils/ByteReader.scala

```diff
@@ -145,7 +153,7 @@ class ByteBufferReader(buf: ByteBuffer) extends ByteReader {
 
   @inline override def getOption[T](getValue: => T): Option[T] = buf.getOption(getValue)
   @inline override def getType(): SType = TypeSerializer.deserialize(this)
-  @inline override def getValue(): SValue = buf.getValue
+  @inline override def getValue(): SValue = ValueSerializer.deserialize(this)
```

Comment:
@greenhat let's move this in extension method to avoid Sigma dependency

Reply:
@ergomorphic I already doing it is the separate branch (not public yet) and it turned out quite "invasive" (little changes in a lot of files). I'd prefer to keep it in a separate PR, which I plan to do after this PR is done.